### ISSUE TITLE
Questions about the masking

### ISSUE THREAD

**Issue by iezhanqingran** [andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning#54]
Hi

Thanks for your great project.

From the paper Mockingjay, I noticed that `the networks to reconstruct the corresponding linear
spectrogram from the masked input` so I assume the reconstruction loss is only computed on masked frames.
So did you try to reconstruct all the frames?

Thanks in advance.

**Comment by andi611** [comment]
Hi,

In the BERT paper, only the masked tokens are predicted.
Following BERT, we only reconstruct the masked frames.

I did not try to reconstruct all the frames, however, this can be easily done by setting this [line](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/61e9edff0e0b7f5266f038bb096265f2bc392ffa/transformer/mam.py#L115) to `torch.ones_like`.

Reconstructing only the masked part makes training more efficient, 
I think reconstructing all will also work but with more update steps.

ANdy

**Comment by iezhanqingran** [comment]
Thanks for your reply, I will look at it!
